File: app/src/main/java/knf/kuma/seeing/SeeingActivity.java
package knf.kuma.seeing;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;

import java.util.List;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.Observer;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.commons.EAHelper;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.SeeingObject;

public class SeeingActivity extends AppCompatActivity {
    @BindView(R.id.toolbar)
    Toolbar toolbar;
    @BindView(R.id.recycler)
    RecyclerView recyclerView;
    @BindView(R.id.error)
    View error;
    private SeeingAdapter adapter;
    private boolean internalMove = false;
    private boolean isFisrt = true;

    public static void open(Context context) {
        context.startActivity(new Intent(context, SeeingActivity.class));
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        setTheme(EAHelper.getTheme(this));
        super.onCreate(savedInstanceState);
        setContentView(getLayout());
        ButterKnife.bind(this);
        toolbar.setTitle("Siguiendo");
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(false);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        adapter = new SeeingAdapter(this);
        recyclerView.setAdapter(adapter);
        ItemTouchHelper touchHelper = new ItemTouchHelper(new ItemTouchHelper.SimpleCallback(ItemTouchHelper.RIGHT, ItemTouchHelper.RIGHT) {
            @Override
            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
                return false;
            }

            @Override
            public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
                final int position = viewHolder.getAdapterPosition();
                final SeeingObject seeingObject = adapter.list.get(position);
                internalMove=true;
                adapter.remove(position);
                final Snackbar snackbar = Snackbar.make(recyclerView, "Anime dropeado", Snackbar.LENGTH_LONG);
                snackbar.setAction("Deshacer", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        error.post(new Runnable() {
                            @Override
                            public void run() {
                                error.setVisibility(View.GONE);
                            }
                        });
                        snackbar.dismiss();
                        internalMove=true;
                        adapter.undo(seeingObject, position);
                    }
                });
                snackbar.show();
            }
        });
        touchHelper.attachToRecyclerView(recyclerView);
        CacheDB.INSTANCE.seeingDAO().getAll().observe(this, new Observer<List<SeeingObject>>() {
            @Override
            public void onChanged(@Nullable List<SeeingObject> list) {
                error.setVisibility(list.size() == 0 ? View.VISIBLE : View.GONE);
                if (!internalMove) {
                    adapter.update(list);
                    if (isFisrt) {
                        isFisrt = false;
                        recyclerView.scheduleLayoutAnimation();
                    }
                }else {
                    internalMove=false;
                }
            }
        });
    }

    @LayoutRes
    private int getLayout(){
        if (PreferenceManager.getDefaultSharedPreferences(this).getString("lay_type","0").equals("0")){
            return R.layout.activity_seening;
        }else {
            return R.layout.activity_seening_grid;
        }
    }
}

File: app/src/main/java/knf/kuma/seeing/SeeingActivity.kt
package knf.kuma.seeing

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import knf.kuma.R
import knf.kuma.commons.EAHelper
import knf.kuma.commons.bind
import knf.kuma.custom.GenericActivity

class SeeingActivity : GenericActivity() {
    val toolbar: Toolbar by bind(R.id.toolbar)
    val tabs: TabLayout by bind(R.id.tabs)
    val pager: ViewPager by bind(R.id.pager)

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(EAHelper.getTheme())
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_seening)
        toolbar.title = "Siguiendo"
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(false)
        toolbar.setNavigationOnClickListener { finish() }
        pager.adapter = SeeingPagerAdapter(supportFragmentManager)
        pager.offscreenPageLimit = 5
        tabs.setupWithViewPager(pager)
        tabs.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabReselected(tab: TabLayout.Tab?) {
                ((pager.adapter as? SeeingPagerAdapter)?.fragmentList)?.let { it[pager.currentItem].onSelected() }
            }

            override fun onTabUnselected(p0: TabLayout.Tab?) {

            }

            override fun onTabSelected(p0: TabLayout.Tab?) {

            }
        })
        pager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrollStateChanged(state: Int) {

            }

            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                ((pager.adapter as? SeeingPagerAdapter)?.fragmentList)?.let { it[position].clickCount = 0 }
            }
        })
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_seeing_auto, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            R.id.auto -> FavToSeeing.onConfirmation(this)
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {

        fun open(context: Context) {
            context.startActivity(Intent(context, SeeingActivity::class.java))
        }
    }
}
